contracts/BalanceLib.sol

```diff
@@ -0,0 +1,146 @@
+/*
+ * Hubii Striim
+ *
+ * Compliant with the Hubii Striim specification v0.12.
+ *
+ * Copyright (C) 2017-2018 Hubii AS
+ */
+
+pragma solidity ^0.4.24;
+
+import {SafeMathInt} from "./SafeMathInt.sol";
+
+library BalanceLib {
+    using SafeMathInt for int256;
+
+    struct Deposit {
+        int256 amount;
+        uint256 timestamp;
+        address token;      //0 for ethers
+        uint256 nonfungible_id;
+    }
+
+    struct Withdrawal {
+        int256 amount;
+        uint256 timestamp;
+        address token;      //0 for ethers
+        uint256 nonfungible_id;
+    }
+
+    struct Balance {
+        int256 ethers;
+        mapping(address => mapping(uint256 => int256)) tokens;
```

Comment:
In light of my comment above wouldn't it be possible to remove Balance' `ethers` and rather store ether values in Balance' `tokens[address(0)][0]`, also suggested by #125.

Reply:
I agree